source/blog/2015-03-27-ember-1-11-0-released.md

```diff
@@ -0,0 +1,245 @@
+
+```handlebars
+<div class="{{color}}"></div>
+```
+
+The inline if helper can also be used in these contexts:
+
+```handlebars
+<div class="{{color}} {{if isEnabled 'active' 'disabled'}}"></div>
+```
+
+For some attributes, like the `disabled` boolean, passing a literal value
+is desirable. An example:
+
+```handlebars
+<input disabled={{isDisabled}}>
+```
+
+To allow the data-binding of non-string values to boolean properties and
+custom element properties, bound attributes are implemented with a
+**property-first** setter.
+
+When binding to an attribute, Ember first checks to see if that attribute is a
+property of the element's DOM node (with normalization of capitalization). If it
+is, the value is set with a property. For example:
+
+```handlebars
+<input disabled={{isDisabled}}>
+```
+
+```js
+// disabled is a property of input elements, so...
+input.disabled = true;
+```
+
+If the attribute is not present as a property, then its value is set as an
+attribute:
+
+```handlebars
+<div class={{color}}>
+```
+
+```js
+// class is not a property of div elements, do...
+div.setAttribute('class', 'red');
+```
+
+For SVG attributes and the style attribute, we've made a exceptions to this pattern
+and simply use `setAttribute` at all times.
+Despite these exceptions, the property-first rule is a good way to describe what is
+happening behind the scenes. In practice, the binding syntax feels natural.
+
+Many thanks to [@mixonic](http://twitter.com/mixonic), [@\_mmun](http://twitter.com/_mmun),
+and [@wycats](http://twitter.com/wycats) for their effort on the design and implementation
+of this feature.
+
+#### Escaping Content in HTMLBars
+
+Bound attribute syntax introduces several new uses of mustaches
+(the `{{` syntax used in Ember templates). These new uses
+come with new security considerations.
+
+Two notable new considerations are mustache use inside style
+contexts, and inside JavaScript contexts. For example, without
+escaping this
+binding would be vulnerable to an XSS attack (via ActiveX
+controls) in IE8:
+
+```handlebars
+<div style="width: {{userProvidedWidth}}px;"></div>
+```
+
+Implementing CSS and JavaScript context-sensitive escaping
+will require further research and development. In Ember 1.11,
+bound attribute syntax will warn in places where we cannot
+safely escape the value.
+
+The examples that follow are intended to demonstrate how this
+works in practice. For example these bindings Just Work:
+
+```handlebars
+{{! Works as expected }}
+<a class="{{someProperty}}"></a>
+
+{{! Works as expected, and escapes unsafe urls }}
+<a href="{{someProperty}}"></a>
+```
+
+However these bindings do not currently have an escaping strategy,
+and thus will warn in the console when used:
+
+```handlebars
+{{! No escaping strategy, log a warning }}
+<a style="width: {{someProperty}}px"></a>
+
+{{! No escaping strategy, log a warning }}
+<a onmouseover="alert('{{someProperty}}');"></a>
```

Comment:
In reality, only `style` warns. This should be made explicit in the blog post, and the implication that JS bindings warn in 1.11 removed. However, we will likely introduce warnings in 1.12.

/cc @rwjblue @stefanpenner @tomdale